spicy/toolchain/src/compiler/codegen/unit-builder.cc

```diff
@@ -175,9 +176,11 @@ Type CodeGen::compileUnit(const type::Unit& unit, bool declare_only) {
     add_hook("0x25_error", {});
     add_hook("0x25_print", {});
     add_hook("0x25_finally", {});
-    add_hook("0x25_confirmed", {});
-    add_hook("0x25_rejected", {});
-    add_hook("0x25_synced", {});
+
+    auto attr_sync = AttributeSet({Attribute("&always-emit")});
+    add_hook("0x25_confirmed", {}, attr_sync);
+    add_hook("0x25_rejected", {}, attr_sync);
+    add_hook("0x25_synced", {}, attr_sync);
```

Comment:
Looks like something I put in while developing (e.g., before I actually invoked the hooks). This is not needed anymore now.

Reply:
We currently actually still need this attribute; otherwise if a unit does not implement the hook the default implementation is removed.

I tried to make this work by annotating these hooks and the field `__trial_mode` with a feature flag and also annotating the args to `confirm` and `reject` in `spicy_rt.hlt`, but it looks like this is still not enough for the optimizer to not remove the default impls.